components/brave_sync/client/bookmark_change_processor.cc

```diff
@@ -909,8 +916,55 @@ void BookmarkChangeProcessor::MigrateOrders() {
   sync_prefs_->SetMigratedBookmarksVersion(1);
 }
 
-void BookmarkChangeProcessor::SendUnsynced(
-    base::TimeDelta unsynced_send_interval) {
+namespace {
+
+static const std::vector<int> g_exponential_waits = {0, 10, 20, 40, 80};
+static const int g_max_send_retries = g_exponential_waits.size() - 1;
+
+int GetCurrentRetryNumber(const bookmarks::BookmarkNode* node,
+  const std::string& last_send_time) {
+  std::string send_retry_number;
+  node->GetMetaInfo("send_retry_number", &send_retry_number);
+  if (send_retry_number.empty()) {
+    // If last_send_time is not empty, then we upgraded app, and wait time
+    // was 10 minutes, first step in exponential delays |g_exponential_waits|
+    if (last_send_time.empty()) {
```

Comment:
why should we even consider the case of `last_send_time.empty()`? if it is empty, we don't even check resend interval

Reply:
Indeed, I agree, no sense to check it in `int GetCurrentRetryNumber()`. Fixed.